flink-runtime/src/test/java/org/apache/flink/runtime/state/UnexpectedStateHandleExceptionTest.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.state;
+
+import org.junit.Test;
+
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.junit.Assert.assertThat;
+
+/**
+ * Tests for {@link UnexpectedStateHandleException}.
+ */
+public class UnexpectedStateHandleExceptionTest {
```

Comment:
Rather then these no-op tests, I'd rather see an ITCase where we actually save with rocksdb and read with file state backend.

Reply:
I added these tests to check whether the somewhat longer string actually renders correctly. And I did find a formatting problem. 😅

I don't want to add a more complicated ITCase now but someone could definitely do it.

Reply:
Hm, I'm not sure what the value of the current tests are. If I add a linebreak to the message formatting at some point, I'd need to touch both tests. 

But I also don't insist on the ITCase: Since you didn't change anything on the exception logic (just using a subclass), I'm fine to leave as is.